### PR TITLE
Ref00320: Get restaurant

### PR DESCRIPTION
**Pull request by ducbet**

### CODE REVIEW

File: app/src/main/java/com/framgia/awesomecity/data/source/local/RestaurantPersistenceContract.java [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+/**
+ * Created by lamvu on 8/14/2017.
+ */
+
+/**
+ * The contract used for the db to save the dishes locally.
+ */
+public final class RestaurantPersistenceContract {
+    /* Inner class that defines the table contents */
+    public static class RestaurantEntry implements BaseColumns {
```
Comment (DoanVanToan) [inline app/src/main/java/com/framgia/awesomecity/data/source/local/RestaurantPersistenceContract.java]: WARNING: Missing a Javadoc comment.

File: app/src/main/java/com/framgia/awesomecity/data/source/RestaurantDataSource.java [hunk 2]
```diff
@@ -3,6 +3,7 @@
 import com.framgia.awesomecity.data.model.Restaurant;
 
 import io.reactivex.Observable;
+import io.reactivex.Single;
```
Comment (DoanVanToan) [inline app/src/main/java/com/framgia/awesomecity/data/source/RestaurantDataSource.java]: ERROR: Unused import: io.reactivex.Single. Use Ctrl+Shift+O (Eclipse) or Ctrl+Alt+O (Intellij) to sort imports

File: app/src/main/java/com/framgia/awesomecity/data/source/remote/RestaurantRemoteDataSource.java [hunk 3]
```diff
@@ -1,5 +1,7 @@
 package com.framgia.awesomecity.data.source.remote;
 
+import android.util.Log;
```
Comment (DoanVanToan) [inline app/src/main/java/com/framgia/awesomecity/data/source/remote/RestaurantRemoteDataSource.java]: ERROR: Unused import: android.util.Log. Use Ctrl+Shift+O (Eclipse) or Ctrl+Alt+O (Intellij) to sort imports

File: app/src/main/java/com/framgia/awesomecity/screen/login/LoginPresenter.java [hunk 4]
```diff
@@ -23,12 +24,16 @@
...
 final class LoginPresenter extends BaseObservable implements LoginContract.Presenter {
     private static final String TAG = LoginPresenter.class.getName();
-
     private final LoginContract.ViewModel mViewModel;
     private CompositeDisposable mCompositeDisposable = new CompositeDisposable();
+    private LoginService mLoginService;
+    private RestaurantRepository mRestaurantRepository;
 
-    public LoginPresenter(LoginContract.ViewModel viewModel) {
+    public LoginPresenter(LoginContract.ViewModel viewModel,
```
Comment (DoanVanToan) [inline app/src/main/java/com/framgia/awesomecity/screen/login/LoginPresenter.java]: ERROR: Redundant 'public' modifier.

File: app/src/main/java/com/framgia/awesomecity/data/source/RestaurantDataSource.java (hunk 2 above)
Comment (DoanVanToan) [inline app/src/main/java/com/framgia/awesomecity/data/source/RestaurantDataSource.java]: WARNING: [UnusedImports] Avoid unused imports such as 'io.reactivex.Single'
https://pmd.github.io/pmd-5.5.1/pmd-java/rules/java/imports.html#UnusedImports

File: app/src/main/java/com/framgia/awesomecity/data/source/remote/RestaurantRemoteDataSource.java (hunk 3 above)
Comment (DoanVanToan) [inline app/src/main/java/com/framgia/awesomecity/data/source/remote/RestaurantRemoteDataSource.java]: WARNING: [UnusedImports] Avoid unused imports such as 'android.util.Log'
https://pmd.github.io/pmd-5.5.1/pmd-java/rules/java/imports.html#UnusedImports

File: app/src/main/java/com/framgia/awesomecity/data/source/RestaurantRepository.java [hunk 5]
```diff
@@ -1,30 +1,40 @@
...
     @Override
     public Observable<Restaurant> getRestaurant() {
-        return mRestaurantRemoteDataSource.getRestaurant();
+        Observable<Restaurant> mLocalRestaurantObservable =
+            mRestaurantLocalDataSource.getRestaurant();
+        Observable<Restaurant> mRemoteRestaurantObservable =
+            mRestaurantRemoteDataSource.getRestaurant();
+        return mRemoteRestaurantObservable;
+        // TODO: 29/08/2017 get restaurant from local or remote
     }
```
Comment (ducbet) [inline app/src/main/java/com/framgia/awesomecity/data/source/RestaurantRepository.java]: đoạn này hôm trước concat.filter.first.asObservable nó cứ ra null mà không nhảy vào filter anh ạ. Nên nay e push trước cái này. Mai em đến hỏi anh sau ạ

**Pull request merged**